l1-contracts/src/core/messagebridge/Inbox.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {MessageBox} from "./MessageBox.sol";
+
+/**
+ * @title Inbox
+ * @author Aztec Labs
+ * @notice Lives on L1 and is used to pass messages into the rollup, e.g., L1 -> L2 messages.
+ */
+contract Inbox is MessageBox {
+  error Inbox__NotPastDeadline();
+  error Inbox__PastDeadline();
+  error Inbox__Unauthorized();
+
+  /**
+   * @dev  struct for sending messages from L1 to L2
+   * @param sender - The sender of the message
+   * @param recipient - The recipient of the message
+   * @param content - The content of the message (application specific) padded to bytes32 or hashed if larger.
+   * @param secretHash - The secret hash of the message (make it possible to hide when a specific message is consumed on L2)
+   * @param deadline - The deadline to consume a message. Only after it, can a message be cancalled.
+   * @param fee - The fee provided to sequencer for including the entry
+   */
+  struct L1ToL2Msg {
+    L1Actor sender;
+    L2Actor recipient;
+    bytes32 content;
+    bytes32 secretHash;
+    uint256 deadline;
+    uint256 fee;
+  }
+
+  event MessageAdded(
+    bytes32 indexed entryKey,
+    address indexed sender,
+    bytes32 indexed recipient,
+    uint256 senderChainId,
+    uint256 recipientVersion,
+    uint256 deadline,
+    uint256 fee,
+    bytes32 content
+  );
+
+  mapping(address account => uint256 balance) public feesAccrued;
+
+  /// @notice Given a message, computes an entry key for the Inbox
+  function computeMessageKey(L1ToL2Msg memory message) public pure returns (bytes32) {
```

Comment:
Should use the `mod p` method while we are using that for sha256 to field. To be replaced later on.